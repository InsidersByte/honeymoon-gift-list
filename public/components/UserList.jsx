/* @flow */

import React from 'react';
import { Paper, Toolbar, ToolbarGroup, ToolbarTitle, IconButton } from 'material-ui';
import PersonAdd from 'material-ui/svg-icons/social/person-add';
import Loader from './Loader';
import ActiveUserList from './ActiveUserList';
import InvitedUserList from './InvitedUserList';

type PropsType = {
    loading: boolean,
    activeUsers: Array<{
        id: number,
        name: string,
        email: string,
    }>,
    invitedUsers: Array<{
        id: number,
        email: string,
    }>,
    loggedInUser: {
        email: string,
    },
    onAdd: Function,
    onDelete: Function,
};

export default function UserList({ loading, activeUsers, invitedUsers, loggedInUser, onAdd, onDelete }: PropsType) {
    return (
        <Paper>
            <Toolbar>
                <ToolbarGroup>
                    <ToolbarTitle text="Users" />
                </ToolbarGroup>

                <ToolbarGroup>
                    <IconButton touch onClick={onAdd}>
                        <PersonAdd />
                    </IconButton>
                </ToolbarGroup>
            </Toolbar>

            <Loader loading={loading}>
                <InvitedUserList
                    users={invitedUsers}
                    onDelete={onDelete}
                />

                <ActiveUserList
                    users={activeUsers}
                    loggedInUser={loggedInUser}
                    onDelete={onDelete}
                />
            </Loader>
        </Paper>
    );
}
