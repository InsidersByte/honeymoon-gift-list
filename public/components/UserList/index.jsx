/* @flow */

import React from 'react';
import { Paper, Toolbar, ToolbarGroup, ToolbarTitle, IconButton } from 'material-ui';
import PersonAdd from 'material-ui/svg-icons/social/person-add';
import Loader from '../Loader';
import ActiveUserList from '../ActiveUserList';
import InvitedUserList from '../InvitedUserList';
import type { UsersType, UserType, AuthUser } from '../../types';

type PropsType = {
  loading: boolean,
  activeUsers: UsersType,
  invitedUsers: UsersType,
  loggedInUser: AuthUser,
  onAdd: () => void,
  onDelete: (user: UserType) => void,
};

const UserList = ({ loading, activeUsers, invitedUsers, loggedInUser, onAdd, onDelete }: PropsType) => (
  <Paper>
    <Toolbar>
      <ToolbarGroup>
        <ToolbarTitle text="Users" />
      </ToolbarGroup>

      <ToolbarGroup>
        <IconButton touch onClick={onAdd}>
          <PersonAdd />
        </IconButton>
      </ToolbarGroup>
    </Toolbar>

    <Loader loading={loading}>
      <InvitedUserList users={invitedUsers} onDelete={onDelete} />

      <ActiveUserList users={activeUsers} loggedInUser={loggedInUser} onDelete={onDelete} />
    </Loader>
  </Paper>
);

export default UserList;
