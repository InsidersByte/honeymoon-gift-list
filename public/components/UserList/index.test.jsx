import React from 'react';
import { shallow } from 'enzyme';
import UserList from './';

const onAdd = jest.fn();
const onDelete = jest.fn();

const props = {
  loading: true,
  activeUsers: [{ id: 1 }],
  invitedUsers: [{ id: 2 }],
  loggedInUser: { email: 'test@test.com' },
  onAdd,
  onDelete,
};

describe('UserList', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<UserList {...props} />);

    expect(wrapper).toMatchSnapshot();
    expect(onAdd).not.toHaveBeenCalled();
    expect(onDelete).not.toHaveBeenCalled();
  });
});
