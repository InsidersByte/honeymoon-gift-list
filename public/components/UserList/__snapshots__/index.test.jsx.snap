// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList should render correctly 1`] = `
<Paper
  circle={false}
  rounded={true}
  transitionEnabled={true}
  zDepth={1}
>
  <Toolbar
    noGutter={false}
  >
    <ToolbarGroup
      firstChild={false}
      lastChild={false}
    >
      <ToolbarTitle
        text="Users"
      />
    </ToolbarGroup>
    <ToolbarGroup
      firstChild={false}
      lastChild={false}
    >
      <IconButton
        disableTouchRipple={false}
        disabled={false}
        iconStyle={Object {}}
        onClick={[Function]}
        tooltipPosition="bottom-center"
        touch={true}
      >
        <SocialPersonAdd />
      </IconButton>
    </ToolbarGroup>
  </Toolbar>
  <Loader
    loading={true}
  >
    <InvitedUserList
      onDelete={[Function]}
      users={
        Array [
          Object {
            "id": 2,
          },
        ]
      }
    />
    <ActiveUserList
      loggedInUser={
        Object {
          "email": "test@test.com",
        }
      }
      onDelete={[Function]}
      users={
        Array [
          Object {
            "id": 1,
          },
        ]
      }
    />
  </Loader>
</Paper>
`;
